datalake_api/querier.py

```diff
@@ -164,6 +164,10 @@ def _unpack(self, result):
         for extra in ['create_time', 'size']:
             if extra in result:
                 r[extra] = result[extra]
+
+        # make sure metadata gets an 'end' key
+        if 'end' not in r['metadata']:
+            r['metadata']['end'] = None
```

Comment:
again, consider `r['metadata'].setdefault('end', None)` (I checked, and there's still a double lookup, but at least it's in C)

Reply:
👍